src/wp-content/themes/twentytwentyone/functions.php

```diff
@@ -447,23 +447,46 @@ function twenty_twenty_one_scripts() {
 		wp_enqueue_script(
 			'twenty-twenty-one-primary-navigation-script',
 			get_template_directory_uri() . '/assets/js/primary-navigation.js',
-			array( 'twenty-twenty-one-ie11-polyfills' ),
+			array(),
 			wp_get_theme()->get( 'Version' ),
-			array( 'in_footer' => true )
+			array(
+				'in_footer' => false,
+				'strategy'  => 'defer',
+			)
 		);
 	}
-
-	// Responsive embeds script.
-	wp_enqueue_script(
-		'twenty-twenty-one-responsive-embeds-script',
-		get_template_directory_uri() . '/assets/js/responsive-embeds.js',
-		array( 'twenty-twenty-one-ie11-polyfills' ),
-		wp_get_theme()->get( 'Version' ),
-		array( 'in_footer' => true )
-	);
```

Comment:
I found this script was being enqueued unconditionally, even when it wasn't needed. So I introduced `twentytwentyone_enqueue_responsive_embeds_script_if_needed` below to only enqueue if the content contains an `iframe`.

Reply:
This is unrelated to the main change here. While I think some of the JS load changes (e.g. using `DOMContentLoaded`) make sense in the scope of this ticket/PR, this change IMO goes too far and should be part of its own ticket, since it belongs into its own conversation.

I'm not opposed, but we shouldn't use this ticket as a general "let's fix bad performance things in bundled themes" :)